src/OpenSpaceToolkit/Astrodynamics/Trajectory/LocalOrbitalFrameTransformProvider.cpp

```diff
@@ -119,18 +116,20 @@ String LocalOrbitalFrameTransformProvider::StringFromType(const LocalOrbitalFram
     return String::Empty();
 }
 
-Transform LocalOrbitalFrameTransformProvider::generateTransform(
-    const LocalOrbitalFrameTransformProvider::Type& aType,
-    const Instant& anInstant,
-    const Vector3d& aPosition,
-    const Vector3d& aVelocity
+Transform LocalOrbitalFrameTransformProvider::GenerateTransform(
+    const LocalOrbitalFrameTransformProvider::Type& aType, const State& aState
 )
 {
+    const Instant& instant = aState.accessInstant();
+    const Vector3d positionCoordinates = aState.getPosition().accessCoordinates();
+    const Vector3d velocityCoordinates = aState.getVelocity().accessCoordinates();
```

Comment:
I think we should explicitly put them in GCRF just in case, no? While we work with raw coordinates below that are assumed to be in GCRF
```suggestion
    const State& stateInGCRF = aState.inFrame(Frame::GCRF())
    const Vector3d positionCoordinates = stateInGCRF.getPosition().accessCoordinates();
    const Vector3d velocityCoordinates = stateInGCRF.getVelocity().accessCoordinates();
```